gh-141004: document Py_HUGE_VAL/IS_FINITE/IS_INFINITE/IS_NAN (#141544)
python/cpython: Doc/c-api/float.rst

   the C11 standard ``<math.h>`` header.

.. c:macro:: Py_HUGE_VAL

>    Same as :c:macro:`Py_INFINITY`.

Reviewer: Up to you, but I would say "equivalent to".

>    Equivalent to :c:macro:`Py_INFINITY`.

Reviewer 2: Py_INFINITY is soft deprecated, so maybe:

```suggestion
   Equivalent to :c:macro:`!INFINITY`.
```

   .. deprecated:: 3.14

>       The macro is soft deprecated.

Reviewer: Could you link to the glossary (i.e. ``` :term:`soft deprecated` ```)?

      return PyFloat_FromDouble(copysign(INFINITY, sign));

> .. c:macro:: Py_IS_FINITE

Reviewer 2: ```suggestion
.. c:macro:: Py_IS_FINITE(X)
```

>    Determines if the given floating-point number has finite value,

Author: ```suggestion
   Determines if the given floating-point number *X* has finite value,
```

>    normal, subnormal or zero, but not infinite or NaN.

Reviewer 3: ```suggestion
   Determines if the given floating-point number has finite value,
   that is, it is normal, subnormal or zero, but not infinite or NaN.
```
[Per the Docs Style guide.](https://devguide.python.org/documentation/style-guide/#use-simple-language)

```diff
- i.e.
+ , that is,
```

>    that is, it is normal, subnormal or zero, but not infinite or NaN.

Reviewer 3: ```suggestion
   Return ``1`` if the given floating-point number *X* is finite,
   that is, it is normal, subnormal or zero, but not infinite or NaN;
   else ``0``.
```

- I find the "Return X ... else X" format more widespread and informative, WDYT?
- "has finite value" -> "is finite"

   .. deprecated:: 3.14
      The macro is :term:`soft deprecated`.  Use :c:macro:`!isfinite` instead.

> .. c:macro:: Py_IS_INFINITE

Reviewer 3: This macro does not exist? I presume you meant:

```suggestion
.. c:macro:: Py_IS_INFINITY
```

> .. c:macro:: Py_IS_INFINITE

Reviewer 2: ```suggestion
.. c:macro:: Py_IS_INFINITY(X)
```

>    Determines if the given floating-point number is positive or negative

Author: ```suggestion
   Determines if the given floating-point number *X* is positive or negative
```

>    infinity.

Reviewer 3: ```suggestion
   Return ``1`` if the given floating-point number *X* is positive or negative
   infinity, else ``0``.
```

   .. deprecated:: 3.14
      The macro is :term:`soft deprecated`.  Use :c:macro:`!isinf` instead.

> .. c:macro:: Py_IS_NAN

Reviewer 2: ```suggestion
.. c:macro:: Py_IS_NAN(X)
```

>    Determines if the given floating-point number is a not-a-number (NaN) value.

Author: ```suggestion
   Determines if the given floating-point number *X* is a not-a-number (NaN) value.
```

>    Determines if the given floating-point number *X* is a not-a-number (NaN) value.

Reviewer 3: ```suggestion
   Return ``1`` if the given floating-point number *X* is a not-a-number (NaN) value,
   else ``0``.
```

      return PyFloat_FromDouble(copysign(INFINITY, sign));

.. c:macro:: Py_IS_FINITE(X)

   Return ``1`` if the given floating-point number *X* is finite,
   that is, it is normal, subnormal or zero, but not infinite or NaN;

>    else ``0``.

Reviewer 2: To me, "else 0" is a little bit unclear. I suggest instead:

```suggestion
   that is, it is normal, subnormal or zero, but not infinite or NaN.
   Return ``0`` otherwise.
```

Same suggestion for Py_IS_INFINITY() and Py_IS_NAN().
